### PR TITLE
chore: adding vsi unit tests

### PR DESCRIPTION
**Pull request by christiancompton**

### CODE REVIEW

**Comment by christiancompton** [comment]
@dingwilson Please let me know if there are other attributes or files we should check for.

File: test/lib/common_test.js [hunk 1]
```diff
@@ -83,6 +83,26 @@ exports.kubernetesFilesGenerator = (applicationName) => [
...
+exports.vsiFiles = [ 'debian/changelog',
+  'debian/compat',
+  'debian/rules',
+  'terraform/main.tf',
+  'terraform/output.tf',
+  'terraform/scripts/build.sh',
+  'terraform/scripts/fetch-state.sh',
+  'terraform/scripts/install.sh',
+  'terraform/scripts/publish-state.sh',
+  exports.jenkinsfile,
```
Comment (dingwilson) [inline test/lib/common_test.js:100]: the Jenkinsfile is not VSI specific (nor does VSI rely on a Jenkinsfile). I'm not certain if it is necessary to include checks on the Jenkinsfile here

File: test/lib/common_test.js [hunk 2]
```diff
@@ -485,6 +505,42 @@ exports.itCreatedKubernetesPipelineFilesWithExpectedContent = function (opts) {
...
+
+  it('terraform start file contains expected content', function () {
+    assert.fileContent(exports.terraformStart, `./${applicationName}`)
+  })
+
+  it('terraform variables file contains expected content', function () {
+    assert.fileContent(exports.terraformVariables, `default = "${applicationName}-01"`)
+  })
+
+  it('terraform variables file contains expected content', function () {
```
Comment (dingwilson) [inline test/lib/common_test.js:539]: oops, accidental copy-paste error... Although not sure if we should keep this test for Jenkinsfile (see other review comment)

**Review by dingwilson — COMMENTED** [review]
Looks great overall - only thing is not sure if we need to include tests for `Jenkinsfile` as part of the VSI integration testing.

**Comment by dingwilson** [comment]
Overall, this covers VSI integration pretty thoroughly. If these integration tests pass, we should be able to safely assume that VSI is working.

We could potentially improve the tests (later on) by checking for Swift-specific strings in the `terraform/scripts/install.sh` and `terraform/scripts/build.sh` files. However, this would have to occur after we iterate on our current implementation for deploying Swift to VSI.

**Pull request closed without merging**